sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamProgress.scala

```diff
@@ -55,6 +55,10 @@ class StreamProgress {
     copied
   }
 
+  private[sql] def toOffset: Offset = {
+    CompositeOffset.fill(currentOffsets.values.toSeq:_*)
```

Comment:
Fixed, this. And also reduce some redundant code in StreamExecution where batchOffset was being constructed by the sink.